File: TrueVote.Services/Database/Izbor.cs
using System;
using System.Collections.Generic;

namespace TrueVote.Services.Database;

public partial class Izbor
{
    public int Id { get; set; }

    public int TipIzboraId { get; set; }

    public DateTime DatumPocetka { get; set; }

    public DateTime DatumKraja { get; set; }

    public string Status { get; set; } = null!;

    public virtual ICollection<Kandidat> Kandidats { get; set; } = new List<Kandidat>();

    public virtual TipIzbora TipIzbora { get; set; } = null!;
}

File: UI/truevote_mobile/lib/models/izbor.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:truevote_mobile/models/tip_izbora.dart';

part 'izbor.g.dart';

@JsonSerializable()
class Izbor {
  int id;
  int? tipIzboraId;
  DateTime? datumPocetka;
  DateTime? datumKraja;
  String? status;
  TipIzbora? tipIzbora;

  Izbor({
    required this.id,
    this.tipIzboraId,
    this.datumPocetka,
    this.datumKraja,
    this.status,
    this.tipIzbora,
  });

  factory Izbor.fromJson(Map<String, dynamic> json) =>
      _$IzborFromJson(json);

  Map<String, dynamic> toJson() => _$IzborToJson(this);
}
